controller/client/client_test.go

```diff
@@ -90,29 +97,37 @@ func TestLoadSave(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	expectedB = true
-	if client.SSLVerify != expectedB {
-		t.Errorf("Expected %t, Got %t", expectedB, client.SSLVerify)
-	}
-
-	expected = "b"
-	if client.Token != expected {
-		t.Errorf("Expected %s, Got %s", expected, client.Token)
-	}
-
-	expected = "c"
-	if client.Username != expected {
-		t.Errorf("Expected %s, Got %s", expected, client.Username)
-	}
-
-	expected = "http://deis.test"
-	if client.ControllerURL.String() != expected {
-		t.Errorf("Expected %s, Got %s", expected, client.ControllerURL.String())
-	}
+	tests = []comparison{
+		comparison{
+			key:      true,
+			expected: client.SSLVerify,
+		},
+		comparison{
+			key:      "b",
+			expected: client.Token,
+		},
+		comparison{
+			key:      "c",
+			expected: client.Username,
+		},
+		comparison{
+			key:      "http://deis.test",
+			expected: client.ControllerURL.String(),
+		},
+		comparison{
+			key:      100,
+			expected: client.ResponseLimit,
+		},
+	}
+
+	checkComparisons(tests, t)
+}
 
-	expectedI = 100
-	if client.ResponseLimit != expectedI {
-		t.Errorf("Expected %d, Got %d", expectedI, client.ResponseLimit)
+func checkComparisons(tests []comparison, t *testing.T) {
+	for _, check := range tests {
+		if check.key != check.expected {
+			t.Errorf("Expected %v, Got %v", check.key, check.expected)
```

Comment:
@Joshua-Anderson this func is great, but it would be better if the caller could call `t.Errorf` instead of here, so that the lines represented in error message reflect where the error actually occurred. thoughts on having this func not take a `*testing.T` param and return an error if `check.key != check.expected`?